### PR TITLE
Add walk-up support and tweak indicator icons

### PR DESCRIPTION
**Pull request by Meorawr**
The note icon that we've been using is a bit confusing. Let's go for a book-with-an-exclamation-mark, because that's obviously a lot clearer for users. This doesn't impact the register list (yet).

Additionally, the size of icons in the bottom row of the tooltip have been increased to make this one asset look good. Also, reuse the old icon to mark profiles that have notes.

image

### CODE REVIEW

**Comment by Solanya** [comment]
Probably needs to at least update the icon used in the directory to be consistent, I'll do it in a few hours if you haven't returned since.

**Comment by Raenore** [comment]
The new icon looks good, but it feels a bit out of place because it does not match the style of the yellow note and the yellow spinner.

I feel that either all of them should be given some sort of consistent modern look, or the new icon should match the yellow style more.

**Comment by Meorawr** [comment]
Yes daddy

**Comment by Raenore** [comment]
A "walkup friendly" was added to this PR (as it also uses a new icon of a smiley to indicate you're happy with people walking up to you.. Or something like that!).

**Status Panel**
| Dashboard | Yes | No |
|  ----    |  ----   |  ----   |
| image | image | image |

**Tooltip**
image

> [!TIP]
> This translates to the "Looking For Contact" option in MRP!
> image

**Comment by Raenore** [comment]
Added the walkup button to profiles

https://github.com/Total-RP/Total-RP-3/assets/172234435/9c10001a-1550-41f9-b7ba-48be77608fa7

This sort of deals with the issue of where people might not necessarily understand what the icon means on the tooltip.

We present it with a hover tooltip on people's profile so they can hover over it and get used to it consciously (or unconsciously) while it hopefully gets added to people's memory that it means "this person doesn't mind walk ups!".

Also fixes the dropdown not showing the icon if you chose "Yes".
image

> [!IMPORTANT]
> This adds one new translation:
> - `REG_PLAYER_RESIDENCE_WALKUP_TT`

**Comment by Raenore** [comment]
The options are now "Yes" or "Do not show".

image

**Comment by Solanya** [comment]
Directory now displays the updated icons for character profiles, and only the new unread icon for companion profiles.

image

While I checked the companion profiles list, I also noticed the header was still called "Master's name", so changed it to be consistent with everywhere else in the addon.

File: totalRP3/Modules/Dashboard/Dashboard.lua [hunk 1]
```diff
@@ -73,6 +74,24 @@ function TRP3_API.dashboard.getCharacterExchangeData()
...
+
+local function switchWalkup()
+	if get("player/character/WU") == TRP3_Enums.WALKUP.NO then
+		onWalkupChange(TRP3_Enums.WALKUP.YES);
+	else
+		onWalkupChange(TRP3_Enums.WALKUP.NO);
+	end
+end
+TRP3_API.dashboard.switchWalkup = switchWalkup;
+
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Dashboard.lua:94]: Not much point in adding these when the Player class has an appropriate setter I feel.

File: totalRP3/Modules/Dashboard/Dashboard.lua [hunk 2]
```diff
@@ -35,6 +35,7 @@ local get, getDefaultProfile = TRP3_API.profile.getData, TRP3_API.profile.getDef
 getDefaultProfile().player.character = {
 	v = 1,
 	RP = TRP3_Enums.ROLEPLAY_STATUS.OUT_OF_CHARACTER,
+	WU =  TRP3_Enums.WALKUP.NO,
```
Comment (Meorawr) [inline totalRP3/Modules/Dashboard/Dashboard.lua:38]:
```suggestion
	WU = TRP3_Enums.WALKUP.NO,
```

**Review by Meorawr — COMMENTED** [review]
Legally I'm not allowed to approve this PR because I own it despite y'all commandeering it for your own sick and twisted needs while I was lain with sickness, so I am forced to instead speak the word "approve".

**Review by Solanya — APPROVED** [review]

**Pull request merged**